tests/tools/kopiarunner/kopia_snapshotter.go

```diff
@@ -282,22 +286,76 @@ func (ks *KopiaSnapshotter) waitUntilServerStarted(ctx context.Context, addr str
 	return nil
 }
 
-// createAndConnectServer creates Repository and a server/client model for interaction
-func (ks *KopiaSnapshotter) createAndConnectServer(serverAddr string, args ...string) (*exec.Cmd, error) {
+// createAndConnectServer creates Repository and a TLS server/client model for interaction
+func (ks *KopiaSnapshotter) createAndConnectServer(serverAddr string, tlsCertFile string, tlsKeyFile string, args ...string) (*exec.Cmd, error) {
 	var cmd *exec.Cmd
 	var err error
 
 	if err = ks.ConnectOrCreateRepo(args...); err != nil {
 		return nil, err
 	}
 
-	if cmd, err = ks.CreateServer(serverAddr); err != nil {
+	serverArgs := append([]string{"--tls-generate-cert", "--tls-cert-file", tlsCertFile, "--tls-key-file", tlsKeyFile})
+	if cmd, err = ks.CreateServer(serverAddr, serverArgs...); err != nil {
 		return nil, err
 	}
 
-	if err = ks.ConnectServer(serverAddr); err != nil {
+	if err = certKeyExist(context.TODO(), DefaultTLSCertPath, DefaultTLSKeyPath); err != nil {
+		return nil, err
+	}
+
+	fingerprint, err := getFingerPrintFromCert(tlsCertFile)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Printf("Identified Server Fingerprint: %v\n", fingerprint)
+
+	serverAddr = fmt.Sprintf("https://%v", serverAddr)
+
+	err = ks.waitUntilServerStarted(context.TODO(), serverAddr, "--server-cert-fingerprint", fingerprint)
+	if err != nil {
+		return cmd, err
+	}
+
+	clientArgs := append([]string{"--server-cert-fingerprint", fingerprint})
+	if err = ks.ConnectServer(serverAddr, clientArgs...); err != nil {
 		return nil, err
 	}
 
 	return cmd, err
 }
+
+func getFingerPrintFromCert(path string) (string, error) {
+	pemData, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	block, rest := pem.Decode([]byte(pemData))
+	if block == nil || len(rest) > 0 {
+		return "", errors.New("pem decoding error")
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return "", err
+	}
+
+	fingerprint := sha256.Sum256(cert.Raw)
+	return hex.EncodeToString(fingerprint[:]), nil
+}
+
+func certKeyExist(ctx context.Context, tlsCertFile, tlsKeyFile string) error {
+	if err := retry.PeriodicallyNoValue(ctx, 1*time.Second, 180, "wait for server start", func() error {
```

Comment:
Readability improvement: if you pull out the `1*time.Second` and `180` magic numbers into named variables or const, this code becomes much easier to read. As is, I have to look into `retry.PeriodicallyNoValue` to figure out what these numbers mean. `"wait for server start"` could also be named something descriptive.